saleor/product/models.py

```diff
@@ -392,6 +393,15 @@ def delete(self, *args, **kwargs):
         super().delete(*args, **kwargs)
 
 
+@receiver(models.signals.post_delete, sender=ProductImage)
```

Comment:
Is there a way not to use signals to implement this?

Reply:
We should cover all places where ProductImage or its parent can be deleted.
Eg. Category/Product/ProductImage delete actions
Plus something can be always deleted from the shell (edge case,but still, most likely during development phase)

Reply:
Do we currently delete the main image? If so, does it also rely on signals? If not, do we want to remove thumbnails?

Reply:
We don't use signals at all, except for one that comes from `django-payments` if I'm correct. Also, we should care about cases when someone does changes using shell.